File: app/src/main/java/info/adavis/sampleapp/jobs/SampleAppJobCreator.java
package info.adavis.sampleapp.jobs;

import com.evernote.android.job.Job;
import com.evernote.android.job.JobCreator;

public class SampleAppJobCreator implements JobCreator
{
    @Override
    public Job create(String tag)
    {
        switch (tag)
        {
            case DemoJob.JOB_TAG:
                return new DemoJob();

            case DemoPeriodicJob.JOB_TAG:
                return new DemoPeriodicJob();
        }

        return null;
    }
}

File: app/src/main/java/info/adavis/sampleapp/jobs/SampleAppJobCreator.kt
package info.adavis.sampleapp.jobs

import com.evernote.android.job.Job
import com.evernote.android.job.JobCreator
import javax.inject.Inject
import javax.inject.Provider

class SampleAppJobCreator @Inject constructor(
        val jobs: @JvmSuppressWildcards Map<String, Provider<Job>>
) : JobCreator {

    override fun create(tag: String): Job? {
        return jobs[tag]?.get()
    }

}
